pkg/service/types.go

```diff
@@ -1,27 +1,59 @@
 // Package service models an instance of a service managed by OSM controller and utility routines associated with it.
 package service
 
-import "fmt"
+import (
+	"fmt"
+	"strings"
+)
 
 const (
 	// namespaceNameSeparator used upon marshalling/unmarshalling MeshService to a string
 	// or viceversa
 	namespaceNameSeparator = "/"
 )
 
+// Locality is the relative locality of a service. ie: if a service is being accessed from the same namespace or a
+// remote cluster.
+type Locality int
+
+const (
+	// LocalNS refers to the local namespace within the local cluster.
+	LocalNS Locality = iota
+	// LocalCluster refers to access within the cluster, but not within the same namespace.
+	LocalCluster
+	// RemoteCluster refers to access from a different cluster.
+	RemoteCluster
+)
+
 // MeshService is the struct defining a service (Kubernetes or otherwise) within a service mesh.
 type MeshService struct {
 	// If the service resides on a Kubernetes service, this would be the Kubernetes namespace.
 	Namespace string
 
 	// The name of the service
 	Name string
+
+	Cluster string
```

Comment:
Wasn't it decide to have this as `ClusterDomain`

Reply:
done!